subsys/usb/device/usb_descriptor.c

```diff
@@ -355,6 +355,32 @@ static void usb_fix_ascii_sn_string_descriptor(struct usb_sn_descriptor *sn)
 	memcpy(sn->bString, runtime_sn, runtime_sn_len);
 }
 
+static void usb_desc_update_mps0(struct usb_device_descriptor *const desc)
+{
+	struct usb_dc_ep_cfg_data ep_cfg = {
+		.ep_addr = 0,
+		.ep_mps = USB_MAX_CTRL_MPS,
+		.ep_type = USB_DC_EP_CONTROL,
+	};
+	int ret;
+
+
+	ret = usb_dc_ep_check_cap(&ep_cfg);
+	if (ret) {
+		/* Try the minimum bMaxPacketSize0 that must be supported. */
+		ep_cfg.ep_mps = 8;
+		ret = usb_dc_ep_check_cap(&ep_cfg);
+		if (ret) {
+			ep_cfg.ep_mps = 0;
+		}
+
+		__ASSERT(ret == 0, "Failed to find valid bMaxPacketSize0");
+	}
+
+	desc->bMaxPacketSize0 = ep_cfg.ep_mps;
+	LOG_ERR("Set bMaxPacketSize0 %u", desc->bMaxPacketSize0);
```

Comment:
LOG_ERR is put here intentionally or here should be LOG_DBG/LOG_INF instead?

Reply:
Thanks, fixed.